app_legacy/Helpers/database/game.php

```diff
@@ -137,24 +150,34 @@ function getGameMetadata(
     }
 
     if ($metrics) {
```

Comment:
It looks like you are are calling the same query twice and doing a union on game and parent but you could just do a single query `where GameID IN (gameID, parentGameID)` though I don't see any performance difference it is mainly just to save some space. Below is how I was able to run it by making a conditional check for the GameID check
```
    if ($metrics) {
        $parentGameId = getParentGameIdFromGameTitle($gameDataOut['Title']);

        $bindings = [
            'gameId' => $gameID,
            'parentGameId' => $parentGameId,
            'achievementType' => $flags,
        ];

    $query = "SELECT aw.HardcoreMode, COUNT(DISTINCT aw.User) as Users
            FROM (
              SELECT awb.User, awb.HardcoreMode
              FROM Awarded AS awb
              LEFT JOIN Achievements AS ach ON ach.ID = awb.AchievementID
              LEFT JOIN UserAccounts AS uab ON uab.User = awb.User
              WHERE ach.GameID IN (:gameId , :parentGameId) AND ach.Flags = :achievementType
              AND (NOT uab.Untracked)
            ) AS aw
            GROUP BY aw.HardcoreMode";

        $gameMetaData = legacyDbFetchAll($query, $bindings);

        $numDistinctPlayersCasual = 0;
        $numDistinctPlayersHardcore = 0;
        foreach ($gameMetaData as $data) {
            if ($data['HardcoreMode'] == UnlockMode::Hardcore) {
                $numDistinctPlayersHardcore = $data['Users'];
            } else {
                $numDistinctPlayersCasual = $data['Users'];
            }
        }

        $gameDataOut['ParentGameID'] = $parentGameId;
        $gameDataOut['NumDistinctPlayersCasual'] = $numDistinctPlayersCasual;
        $gameDataOut['NumDistinctPlayersHardcore'] = $numDistinctPlayersHardcore;
    }

    $gameDataOut['NumAchievements'] = $numAchievements;
```

Reply:
Excellent feedback, this is much more concise. Addressed in https://github.com/RetroAchievements/RAWeb/pull/1530/commits/f367e723499103b84452ce04e5087506bb829f7c